### ISSUE TITLE
verify -provider option processing dependent on parameter order

### ISSUE THREAD

**Issue by baentsch** [openssl/openssl#16732]
When [testing QSC encode/cert generation](https://github.com/open-quantum-safe/oqs-provider/issues/2) thus using `openssl verify` for the first time with our (external) [oqsprovider](https://github.com/open-quantum-safe/oqs-provider), it became apparent that the position of the parameter `-provider` within the `verify` parameter list does make a difference:

This works: `LD_LIBRARY_PATH=.local/lib64 OPENSSL_MODULES=_build/oqsprov .local/bin/openssl verify -provider oqsprovider -provider default -CAfile dilithium2_CA.crt dilithium2_srv.crt` 

This fails: `LD_LIBRARY_PATH=.local/lib64 OPENSSL_MODULES=_build/oqsprov .local/bin/openssl verify -CAfile dilithium2_CA.crt dilithium2_srv.crt  -provider oqsprovider -provider default` in the following way:
```
error 24 at 0 depth lookup: issuer certificate doesn't have a public key
error dilithium2_srv.crt: verification failed
C0C1F3667C7F0000:error:03000072:digital envelope routines:X509_PUBKEY_get0:decode error:crypto/x509/x_pubkey.c:444:
C0C1F3667C7F0000:error:03000072:digital envelope routines:X509_PUBKEY_get0:decode error:crypto/x509/x_pubkey.c:444:
C0C1F3667C7F0000:error:03000072:digital envelope routines:X509_PUBKEY_get0:decode error:crypto/x509/x_pubkey.c:444:
Could not open file or uri for loading certificate file from -provider
[...]
```

-> The initial error is a consequence of `verify` not knowing about QSC keys as the provider has not been loaded. The second error seems to indicate that the `-provider` option is taken as a certificate file. 

This seems to be _very_ similar to https://github.com/openssl/openssl/issues/14069 (fixed by https://github.com/openssl/openssl/pull/14135).

**Comment by levitte** [comment]
Yes, this is correct.

This is probably a choice that we should review more consistently at some point...  we have a variety of commands that treat command line options differently; some option values are processed directly as they are parsed from the command line (left to right), while others are stored away and processed after the whole command line has been parsed.

Neither way of doing things is really a bug in itself, but it can be confusing when they're mixed.

The safer choice is to assume that command line option values are treated directly as they appear on the command line (which I believe is the oldest norm in general unix commands).

**Comment by baentsch** [comment]
>  we have a variety of commands that treat command line options differently
> Neither way of doing things is really a bug in itself, but it can be confusing when they're mixed.

Yes, this _is_ confusing. Shouldn't it be so that the way the options are handled at least should be documented to users? When re-reading https://www.openssl.org/docs/manmaster/man1/openssl-verify.html there is no hint that the order of placement of the `-provider` option makes a significant difference as to the operation of the command.

Thus, may I suggest to open an issue to 
>  This is probably a choice that we should review more consistently at some point [... as] we have a variety of commands that treat command line options differently 

Until that is resolved, may I further suggest to at least add a warning to the Provider section at https://www.openssl.org/docs/manmaster/man1/openssl.html stating something along the lines above, e.g., "Be aware that the placement of these options relative to other command line options in different `openssl` commands makes a difference; as a general rule, if needed, these options should be placed before any other command-specific option in order for them to be honoured."

**Comment by levitte** [comment]
Documentation should probably be altered more generically.  Another option that's processed directly in most commands is `-config` (interestingly enough, not with `openssl ca` and `openssl req`)
